src/coreclr/vm/clsload.cpp

```diff
@@ -305,13 +327,32 @@ TypeHandle ClassLoader::LoadTypeByNameThrowing(Assembly *pAssembly,
     }
     CONTRACT_END
 
-    NameHandle nameHandle(nameSpace, name);
-    if (fLoadTypes == DontLoadTypes)
-        nameHandle.SetTokenNotToLoad(tdAllTypes);
-    if (fNotFound == ThrowIfNotFound)
-        RETURN pAssembly->GetLoader()->LoadTypeHandleThrowIfFailed(&nameHandle, level);
-    else
-        RETURN pAssembly->GetLoader()->LoadTypeHandleThrowing(&nameHandle, level);
+    ClassLoader* classLoader = pAssembly->GetLoader();
```

Comment:
This method is called from number of places. I do not think we want this change in all those other places. It will change the behavior in subtle ways and it may make it impossible to load types with `+` in the type name. `+` in (non-fully qualified) type name is valid and I can imagine things like obfuscators may take advantage of it.